DOC-996 Add schema_evolution.processors (#173)
redpanda-data/rp-connect-docs: modules/components/pages/outputs/snowflake_streaming.adoc

*Type*: `bool`

=== `schema_evolution.processors`

> A series of processors to execute when new columns are added to the Snowflake table. You can use these processors to perform tasks, including:

Reviewer: ```suggestion
A series of processors to execute when new columns are added to the Snowflake table. You can use these processors to:
```

Author: Updated.

> - Running side effects when the schema evolves.

Reviewer: ```suggestion
- Run side effects when the schema evolves.
```

Author: Updated

> - Running side effects when the schema evolves.
> - Enriching the message with additional information to guide the schema changes.

Reviewer: ```suggestion
- Run side effects when the schema evolves.
- Enrich the message with additional information to guide the schema changes.
```

> === `schema_evolution.new_column_type_mapping`

Reviewer 2: This is deprecated now, we can should probably remove it.

Author: Good spot! I've removed this.

  }
```

=== `schema_evolution.processors`

A series of processors to execute when new columns are added to the Snowflake table. You can use these processors to perform tasks, including:

- Running side effects when the schema evolves.
- Enriching the message with additional information to guide the schema changes.

For example, a processor could read the schema from the schema registry that a message was produced with and use that schema to determine the data type of the new column in Snowflake.

The input to these processors is an object with the value and name of the new column, the original message, and details of the Snowflake table the output writes to. 

For example: `{"value": 42.3, "name":"new_data_field", "message": {"existing_data_field": 42, "new_data_field": "db_field_name"}, "db": MY_DATABASE", "schema": "MY_SCHEMA", "table": "MY_TABLE"}`

The metadata remains the same as in the original message that triggered the schema update.

> 

Reviewer 2: additionally the output of these processors should be a message without errors that contains a string with the column type to add for the column in Snowflake.

Author: Updated. Please check for accuracy.
